source/features/add-pr-branches.js

```diff
@@ -0,0 +1,107 @@
+import {h} from 'dom-chef';
+import select from 'select-dom';
+import * as cache from '../libs/cache';
+import * as api from '../libs/api';
+import {getOwnerAndRepo} from '../libs/page-detect';
+
+function buildQuery(owner, repo, numbers) {
+	let query = `
+{
+  repository(owner: "${owner}", name: "${repo}") {`;
+
+	for (const number of numbers) {
+		query += `
+    ${number}: pullRequest(number: ${number.replace('issue_', '')}) {
+      baseRef {
+        name
+        repository {
+          url
+          owner {
+            login
+          }
+        }
+      }
+      headRef {
+        name
+        repository {
+          url
+          owner {
+            login
+          }
+        }
+      }
+    }`;
+	}
+
+	query += `
+  }
+}`;
+	return query;
+}
+
+async function fetchFromApi(owner, repo, numbers) {
+	const query = buildQuery(owner, repo, numbers);
+	const response = await api.v4(query);
+	console.log('add-pr-branches 3:', numbers, response);
+
+	if (response.data && response.data.repository) {
+		const d = {};
+		for (const [id, data] of Object.entries(response.data.repository)) {
+			d[id] = extractBranches(owner, data);
+		}
+		return d;
+	}
+}
+
+function extractBranches(owner, data) {
```

Comment:
Instead of parsing it here, just cache the response and use that in the JSX. You can also create utility functions like this to further reduce repetition:

```js
const shortenLabel = (owner, ref) => (ref.repository.owner.login === owner ? ref.name : `${ref.repository.owner.login}:${ref.name}`);

const getLink = ref => (
    <span class="commit-ref css-truncate user-select-contain">
    	<a title={ref.label} href={`${ref.repository.url}/tree/${ref.name}`}>{ref.label}</a>
    </span>
);
```